### DISCUSSION TITLE
How to change the text of a button when clicked

### DISCUSSION THREAD

**Discussion by Mikachu2333** [emilk/egui#5312] [Q&A] [answered]
## Background
As the title shows, I'd want to make a button to **change and show** the `always on top` status. As the following gif shows, for example, `⚓` means that the program is _always on top_ and `🔱` means normal (this gui is made by another programming language and i want to use rust to re-make it).

## Problem
I knew how to set the `always on top` status form #4366 , but I don't know how to change the button text when the program is running, what should i do?

## What i want
sth like the following "fake" code shows
```
let btn1 = button("Original status (ON TOP)")
if btn1.clicked(){
    if btn1.get_text() == "Original status (ON TOP)".to_owned(){// no need, just to show what i want
        on_top(false);
        
        match btn1.set_text("Status changed (NOT ON TOP)"){
            OK(_)=>msgbox("Success!")),//change on top or not
            Err(e)=>println!("{:?}",e)
        }
    }else{
        todo!("etc.");
    }
}
```

https://github.com/user-attachments/assets/1117d8ab-8660-4a1e-8404-2673951cf55c

**Comment by lucasmerlin** [comment, marked as answer]
You'd do something like this:

```rust
let mut on_top = false;

let text = if on_top {
  "Original status (ON TOP)"
} else {
  "Status changed (NOT ON TOP)"
}

if ui.button(text).clicked() {
  on_top = !on_top;
  set_on_top(on_top);
}
```
